Makefile

```diff
@@ -1,11 +1,55 @@
-# Specify the name for the binaries
-NODE_DISK_MANAGER=ndm
-DOCKER_IMAGE_NAME?= openebs/node-disk-manager
-DOCKER_IMAGE_TAG?= $(subst /,-,$(shell git rev-parse --abbrev-ref HEAD))
+# Build the node-disk-manager image.
 
-# Use this to build only the node-disk-manager.
+build: clean vet fmt ndm version docker
 
-build: header ndm docker
+PACKAGES = $(shell go list ./... | grep -v '/vendor/')
+
+# Determine the arch/os
+XC_OS?= $(shell go env GOOS)
+XC_ARCH?= $(shell go env GOARCH)
+ARCH:=${XC_OS}_${XC_ARCH}
+
+# TAG is the tag of the docker image
+TAG?=$(shell git describe --tags --always)
+
+# IMAGE is the image name of the node-disk-manager docker image.
+IMAGE:=openebs/node-disk-manager-${XC_ARCH}:${TAG}
+
+# The ubuntu:16.04 image is being used as base image.
+BASEIMAGE:=ubuntu:16.04
+
+# Tools required for different make targets or for development purposes
+EXTERNAL_TOOLS=\
+	github.com/golang/dep/cmd/dep \
+	gopkg.in/alecthomas/gometalinter.v1
+
+vet:
+	go list ./... | grep -v "./vendor/*" | xargs go vet
+
+fmt:
+	find . -type f -name "*.go" | grep -v "./vendor/*" | xargs gofmt -s -w -l
+
+# Run the bootstrap target once before trying gometalinter in Development environment
+golint:
+	@gometalinter.v1 --install
+	@gometalinter.v1 --vendor --deadline=600s ./...
+
+version:
+	@echo $(VERSION)
```

Comment:
Where will the value for VERSION be provided?

Reply:
I will remove it !